browser/src/react-redux/App/Signup/SignupForm.jsx

```diff
@@ -0,0 +1,156 @@
+      },
+      buttonDisabled: false
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  handleChange(key, e) {
+    this.setState({
+      [key]: e
+    });
+  }
+  onSubmit(e) {
+    this.setState({
+      errors: {
+        general: "",
+        email: "",
+        password: ""
+      }
+    });
+
+    e.preventDefault();
+
+    const user = {
+      email: this.state.email,
+      password: this.state.password
+    };
+
+    axios
+      .post("/api/user", user)
+      .then(res => {
+        if (res.request.status === 201) {
+          console.log(`Success!: ${JSON.stringify(res.data.user, null, "  ")}`);
+        }
+      })
+      .catch(err => {
+        console.log(
+          `Error!: ${JSON.stringify(err.response.data.error, null, "  ")}`
+        );
+        if (err.response.data.error.errors) {
+          let newErrors = utility.extractErrors(err.response.data.error.errors);
+
+          this.setState({
+            errors: { ...this.state.errors, ...newErrors }
+          });
+        } else {
+          this.setState({
+            errors: {
+              general: err.response.data.error.message
+            }
+          });
+        }
+      });
+  }
+  componentDidUpdate(prevProps, prevState) {
+    //check if password and confirmPassword fields match. If not, set error message to state.
+
+    if (
+      this.state.confirmPassword &&
+      this.state.password != this.state.confirmPassword &&
+      !this.state.errors.confirmPassword
+    ) {
+      this.setState({ errors: { confirmPassword: "Passwords do not match." } });
+    } else if (prevState.errors.confirmPassword) {
+      this.setState({
+        errors: {
+          confirmPassword: ""
+        }
+      });
+    }
+    //disable submit until following requirements are met
+    if (
+      this.state.email &&
+      this.state.password &&
+      this.state.confirmPassword &&
+      this.state.password === this.state.confirmPassword &&
+      prevState.buttonDisabled
+    ) {
+      this.setState({
+        buttonDisabled: false
+      });
+    } else if (
+      this.state.password !== this.state.confirmPassword &&
+      prevState.buttonDisabled == false
+    ) {
+      this.setState({
+        buttonDisabled: true
+      });
+    }
+  }
+  render() {
+    let name = Object.keys(this.state);
+    const error = this.state.errors;
+    return (
+      <div className="signup-form">
+        <form onSubmit={this.onSubmit} className="signup-form__form-group">
+          <h1 className="signup-form__title">Signup for Site Builder</h1>
+          <span className="signup-form__general-error">
+            {this.state.errors.general}
+          </span>
+          <Form
+            label="Email"
+            name={name[0]}
+            value={this.state[0]}
+            handleChange={this.handleChange}
+            type="text"
+            error={error.email}
+          />
+          <Form
```

Comment:
I'm still not convinced you need a separate component for this, but if you feel like you want it, I'm not against it. As we talked about, I would label more appropriately as ..something like.. `<InputGroup>` or something similar.